### ISSUE TITLE
Proxying doesn't work with `mitmproxy`

### ISSUE THREAD

**Issue by miliog** [typst/typst#2294]
### Description

Downloading packages via `compile` fails. For `update`, the proxy isn't being used at all.

![Screenshot from 2023-10-02 22-18-45](https://github.com/typst/typst/assets/146289769/696cd88b-791c-45fb-8330-317cfe41e378)

Status 400 is coming from `mitmproxy`. It doesn't understand the HTTP request for some reason.

Below are all HTTP requests that successfully passed through `mitmproxy`:

![Screenshot from 2023-10-02 22-22-02](https://github.com/typst/typst/assets/146289769/a248cdad-24e6-4b59-bbbd-ed57ae1dc114)

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
@zicklag any ideas?

**Comment by zicklag** [comment]
Not off the top of my head. @miliog is this the proxy you're using [`mitmproxy`](https://github.com/mitmproxy/mitmproxy)?

I _might_ be able to take a quick look at it...

**Comment by laurmaedje** [comment]
Oh.. well, it's an easy fix at least.

**Comment by zicklag** [comment]
> I still don't know why mitmproxy in particular doesn't work even though curl can do it just fine.

Yeah, I think that's the only mystery, now. I can confirm that downloading packages works through a squid proxy, so there must be some nuance in the format of the proxy request that is made by the [`ureq`](https://github.com/algesten/ureq) library that we use for making requests.

I think it might be an issue for the `ureq` crate, if we can double-check with a minimal reproducible example that `ureq` requests just don't go through the `mitmproxy` correctly.

**Comment by laurmaedje** [comment]
I've changed the release fetching to use the agent, too, so the only remaining issue is the mitmproxy thing.
